test/test_testing.py

```diff
@@ -2207,6 +2207,18 @@ def test_no_warning_on_import(self) -> None:
         out = self._check_python_output("import torch")
         self.assertEqual(out, "")
 
+    def test_not_import_sympy(self) -> None:
+        try:
+            self._check_python_output("import torch;import sys;assert 'sympy' not in sys.modules")
+        except subprocess.CalledProcessError:
+            self.fail("PyTorch should not depend on SymPy at import time as importing SymPy is *very* slow.\n"
+                      "See the beginning of the following blog post for how to profile and find which file is importing sympy:\n"
+                      "https://dev-discuss.pytorch.org/t/delving-into-what-happens-when-you-import-torch/1589\n\n"
+                      "If you hit this error, you may want to:\n"
+                      "  - Refactor your code to avoid depending on sympy files you may not need to depend\n"
+                      "  - Use TYPE_CHECKING if you are using sympy + strings if you are using sympy on type annotations\n"
+                      "  - Import things that depend on SymPy locally")
```

Comment:
Why not check the output rather than wait for assert (if Python is invoked with `-OO` asserts will be ignored even in subprocessed, aren't they?)
```suggestion
        out = self._check_python_output("import torch;import sys;print('sympy' not in sys.modules)")
        self.assertEqual(out.strip(), "True", "PyTorch should not depend on SymPy at import time as importing SymPy is *very* slow.\n"
                      "See the beginning of the following blog post for how to profile and find which file is importing sympy:\n"
                      "https://dev-discuss.pytorch.org/t/delving-into-what-happens-when-you-import-torch/1589\n\n"
                      "If you hit this error, you may want to:\n"
                      "  - Refactor your code to avoid depending on sympy files you may not need to depend\n"
                      "  - Use TYPE_CHECKING if you are using sympy + strings if you are using sympy on type annotations\n"
                      "  - Import things that depend on SymPy locally")
```

Reply:
Another reason to prefer `check_process_output` is exceptions handling might hide the problem that `import torch` failed, while `check_output` will assure that script has executed, but returned an unexpected resutl

Reply:
Good point. Changed.